master/docs/manual/cfg-buildsteps.rst

```diff
@@ -1240,8 +1267,65 @@ The arguments are identical to :bb:step:`ShellCommand`.
 ::
 
     from buildbot.plugins import steps
+
     f.addStep(steps.Configure())
 
+.. bb:step:: CMake
+
+CMake
++++++
+
+.. py:class:: buildbot.steps.cmake.CMake
+
+This is intended to handle the :command:`cmake` step for projects that use `CMake-based build systems <http://cmake.org>`_.
+
+.. note::
+
+   Links below point to the CMake v3.5 documentation.
+   Please make sure that you check the documentation for the CMake you use.
+
+In addition to the parameters :bb:step:`ShellCommand` supports, this step accepts the following parameters:
+
+``path``
+    Either a path to a source directory to (re-)generate a build system for it in the current working directory.
+    Or an existing build directory to re-generate its build system.
+
+``generator``
+    A build system generator.
+    See `cmake-generators(7) <https://cmake.org/cmake/help/v3.5/manual/cmake-generators.7.html>`_ for available options.
```

Comment:
I am not sure I understood the question: in CMake a generator is what produces the actual build system.  In my day job, it's Ninja.

Could you please clarify what you mean?